scripts/src/java/org/oppia/android/scripts/common/MavenDependenciesListGenerator.kt

```diff
@@ -0,0 +1,333 @@
+   * in maven_install.json and is also obtained from the Bazel Query.
+   *
+   * @param pathToMavenInstall path to the maven_install.json file
+   * @param bazelQueryDepsNames list of dependency names generated from Bazel Query
+   *
+   * @return list of [MavenListDependency]
+   */
+  fun getDependencyListFromMavenInstall(
+    pathToMavenInstall: String,
+    bazelQueryDepsNames: List<String>
+  ): List<MavenListDependency> {
+    val dependencyTree = retrieveDependencyTree(pathToMavenInstall)
+    return dependencyTree.mavenListDependencies.dependencyList.filter { dep ->
+      omitVersionAndReplaceColonsHyphensPeriods(dep.coord) in bazelQueryDepsNames
+    }
+  }
+
+  /**
+   * Extracts the license names and license links of the dependencies from their corresponding POM
+   * files.
+   *
+   * @param finalDependenciesList list of dependencies that is obtained by the intersection of
+   *    the list generated by Bazel Query and the list generated from maven_install.json
+   *
+   * @return mavenDependencyList that has dependencies with licenses extracted from their POM files
+   */
+  fun retrieveDependencyListFromPom(
+    finalDependenciesList: List<MavenListDependency>
+  ): MavenDependencyList {
+    val mavenDependencyList = finalDependenciesList.map { it ->
+      // Remove ".jar" or ".aar" or any other extension from the specified url.
+      val pomFileUrl = "${it.url?.substringBeforeLast('.')}.pom"
+      val artifactName = it.coord
+      val artifactVersion = artifactName.substringAfterLast(':')
+      val pomFile = licenseFetcher.scrapeText(pomFileUrl)
+      val mavenDependency = MavenDependency.newBuilder().apply {
+        this.artifactName = it.coord
+        this.artifactVersion = artifactVersion.toString()
+        this.addAllLicense(extractLicenseLinksFromPom(pomFile))
+      }
+      mavenDependency.build()
+    }
+    return MavenDependencyList.newBuilder().addAllMavenDependency(mavenDependencyList).build()
+  }
+
+  /**
+   * Parses the maven_install.json file to compile the list of maven dependencies.
+   *
+   * @param pathToMavenInstall path to the maven_install.json file
+   * @param bazelQueryDepsNames list of dependency names obtained from the bazel query
+   *
+   * @return list of [MavenListDependency] that contains the artifact name and a URL that is used
+   *    to obtain the URL of the POM file of the dependency
+   */
+  fun genearateDependenciesListFromMavenInstall(
+    pathToMavenInstall: String,
+    bazelQueryDepsNames: List<String>
+  ): List<MavenListDependency> {
+    val dependencyTree = retrieveDependencyTree(pathToMavenInstall)
+    val mavenInstallDependencyList = dependencyTree.mavenListDependencies.dependencyList
+    return mavenInstallDependencyList.filter { dep ->
+      bazelQueryDepsNames.contains(omitVersionAndReplaceColonsHyphensPeriods(dep.coord))
+    }
+  }
+
+  /**
+   * Helper function to parse the text proto file to a proto class.
```

Comment:
```suggestion
   * Parses the text proto file to a proto class.
```

(thought FWIW KDocs on private methods are optional).

Reply:
Done.